dll/win32/shell32/CDefView.cpp

```diff
@@ -867,11 +867,15 @@ BOOL CDefView::_Sort()
 
 PCUITEMID_CHILD CDefView::_PidlByItem(int i)
 {
+    if (!m_ListView)
```

Comment:
I suspect that this could be hiding another issue here. Under what conditions can it happen? I ask because Initialization of the CDefView is supposed to fail if this is null and this gets destroyed in the destructor of the class.

Reply:
I don't think so. m_ListView is an object (not a pointer) and it is default-initialized.  The listview is "created" in CDefView::CreateList(), which is called from CDefView::OnCreate(), i.e. after the view window was created by IShellView_CreateViewWindow. The problem is an assertion failure that happens during shell32_winetest shlview in test_IFolderView.
```
00 00000000`0012f910 000007ff`a45bbb66     shell32!ATL::CWindow::SendMessageW+0x60 [E:\amd64\reactos\sdk\lib\atl\atlwin.h @ 1112] 
01 00000000`0012f950 000007ff`a45bbc10     shell32!CListView::GetItem+0x26 [E:\amd64\reactos\sdk\include\reactos\ui\rosctrls.h @ 109] 
02 00000000`0012f980 000007ff`a45c7995     shell32!CListView::GetItemData+0x60 [E:\amd64\reactos\sdk\include\reactos\ui\rosctrls.h @ 209] 
03 00000000`0012fa20 000007ff`a45bdf88     shell32!CDefView::_PidlByItem+0x25 [E:\amd64\reactos\dll\win32\shell32\CDefView.cpp @ 871] 
04 00000000`0012fa50 00000001`4007e668     shell32!CDefView::Item+0x28 [E:\amd64\reactos\dll\win32\shell32\CDefView.cpp @ 2819] 
05 00000000`0012fa90 00000001`4007ac48     shell32_winetest!test_IFolderView+0x258 [E:\amd64\reactos\modules\rostests\winetests\shell32\shlview.c @ 747] 
06 00000000`0012fcf0 00000001`400822a9     shell32_winetest!func_shlview+0x28 [E:\amd64\reactos\modules\rostests\winetests\shell32\shlview.c @ 1524] 
07 00000000`0012fd20 00000001`40082193     shell32_winetest!run_test+0xa9 [E:\amd64\reactos\sdk\include\reactos\wine\test.h @ 879] 
```

IFolderView_Item() is supposed to return E_INVALIDARG here, because there is no window and no list yet. CDefView::Item() does exactly this, if _PidlByItem() returns NULL.

Reply:
Hm indeed. This change is perfectly fine. All the rest of the methods of this class probably have the same issue.